File: backend/src/main/java/com/petssocial/entity/Moment.java
package com.petssocial.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("moments")
public class Moment {
    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    private Long userId;

    private Long petId;

    private String content;

    private String images; // JSON array

    private Double longitude;

    private Double latitude;

    private String locationName;

    private Integer likeCount;

    private Integer commentCount;

    private Integer status; // 0: deleted, 1: active

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;
}

File: frontend/src/types/moment.ts
import type { UserInfo } from './auth'
import type { Pet } from './pet'

export interface Moment {
  id: string
  userId: string
  user: UserInfo
  petId?: string
  pet?: Pet
  content: string
  images: string[]
  location?: string
  likeCount: number
  commentCount: number
  liked: boolean
  createdAt: string
}

export interface MomentForm {
  petId?: string
  content: string
  images?: string[]
  location?: string
}
